cmd/kube-controller-manager/app/policy.go

```diff
@@ -49,6 +50,7 @@ func startDisruptionController(ctx context.Context, controllerContext Controller
 		controllerContext.RESTMapper,
 		scaleClient,
 		client.Discovery(),
+		clock.RealClock{},
```

Comment:
In general, I would avoid adding a function that is only used for tests, like `NewDisruptionControllerInternal`, but since we have to add it, I prefer that the non internal function has as much parameters as possible, so usage is more explicit.

Reply:
I see, but isn't clock by nature a technical detail for testing rather than part of the public API one needs to call to construct an object? Also, another consideration is that the PR requires approval from cmd/OWNERS which can be easily avoided.